executor/distsql.go

```diff
@@ -196,6 +196,10 @@ type IndexReaderExecutor struct {
 
 // Close clears all resources hold by current object.
 func (e *IndexReaderExecutor) Close() error {
+	if e.table != nil && e.table.Meta().TempTableType != model.TempTableNone {
```

Comment:
How about implementing a new function for the logic about checking whether the table is a TmpTable?

Reply:
In general, we do not need to check `e.table != nil`. 
It's special here because a test case construct the `IndexReaderExecutor` manually and don't set `e.table` field.

The remain part is `e.table.Meta().TempTableType != model.TempTableNone`.
Define a one line function for that does not seems beneficial: the change will introduce a new exported function without simplifying the code much.